{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5d9273c2-ed44-454a-af85-b911dec7c509",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import dendropy\n",
    "import treeswift\n",
    "from treeswift import read_tree_newick\n",
    "#import matplotlib.pyplot as plt\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a46d15d2-29ff-40dd-8dd6-512425996335",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/revisions_ONT_reads_variable_err\n"
     ]
    }
   ],
   "source": [
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1e846c26-733d-4cad-905b-1fd664580811",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"results\"   # change to your folder\n",
    "\n",
    "# list only directories (not files)\n",
    "dirs = [d for d in os.listdir(path) if os.path.isdir(os.path.join(path, d)) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3cc3a805-ca2b-470c-b2ce-d5aba5366c08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ONT_095_Claded_Chunked',\n",
       " 'ONT_099_Uncladed_Chunked',\n",
       " 'ONT_090_Claded_Chunked',\n",
       " 'ONT_085_Claded_Chunked',\n",
       " 'ONT_099_Claded_Chunked',\n",
       " 'ONT_1_Claded_Chunked',\n",
       " 'ONT_085_Uncladed_Chunked',\n",
       " 'ONT_095_Uncladed_Chunked',\n",
       " 'ONT_1_Uncladed_Chunked',\n",
       " 'ONT_090_Uncladed_Chunked']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dirs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "1b9ad97c-0fd2-48e4-ab60-eefa4a9673d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_path = \"/Users/nora/Documents/ml_metagenomics\"\n",
    "length_folder = \"ONT_read_length\"\n",
    "\n",
    "true_clades = pd.read_csv(os.path.join(my_path, \"clade_targets.txt\"), sep = \" \", header = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "6c34aab6-ef39-4121-8dfc-e07fbbef7fb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>clade</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G001940755</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G001918745</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G001920395</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G001918475</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G000724145</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10570</th>\n",
       "      <td>G001652645</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10571</th>\n",
       "      <td>G001652685</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10572</th>\n",
       "      <td>G900061685</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10573</th>\n",
       "      <td>G900061725</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10574</th>\n",
       "      <td>G900061705</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10575 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           genome  clade\n",
       "0      G001940755     13\n",
       "1      G001918745     13\n",
       "2      G001920395     13\n",
       "3      G001918475     13\n",
       "4      G000724145     13\n",
       "...           ...    ...\n",
       "10570  G001652645      9\n",
       "10571  G001652685      9\n",
       "10572  G900061685      9\n",
       "10573  G900061725      9\n",
       "10574  G900061705      9\n",
       "\n",
       "[10575 rows x 2 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_clades"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "bab2798e-90c1-475d-b544-b488909da2ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ONT_095_Claded_Chunked 2.995 0.936\n",
      "ONT_099_Uncladed_Chunked 3.364 0.0\n",
      "ONT_090_Claded_Chunked 4.307 0.904\n",
      "ONT_085_Claded_Chunked 6.636 0.847\n",
      "ONT_099_Claded_Chunked 2.75 0.947\n",
      "ONT_1_Claded_Chunked 2.574 0.949\n",
      "ONT_085_Uncladed_Chunked 8.125 0.0\n",
      "ONT_095_Uncladed_Chunked 3.626 0.0\n",
      "ONT_1_Uncladed_Chunked 3.31 0.0\n",
      "ONT_090_Uncladed_Chunked 5.093 0.0\n"
     ]
    }
   ],
   "source": [
    "my_big_df = pd.DataFrame()\n",
    "\n",
    "for d in dirs:\n",
    "    \n",
    "    dir_path = os.path.join(path, d)\n",
    "    \n",
    "    if os.path.isdir(dir_path):\n",
    "        # Define the two files you want to read\n",
    "        file1 = os.path.join(dir_path, \"classes.out\")\n",
    "        file2 = os.path.join(dir_path, \"all.pl_err\" )\n",
    "\n",
    "        # Only proceed if both files exist\n",
    "        if os.path.exists(file1) and os.path.exists(file2):\n",
    "            df1 = pd.read_csv(file1, sep=\"\\t\")\n",
    "            \n",
    "            df2 = pd.read_csv(file2, sep=\" \", header = None, names=[\"contigID\", \"empty\", \"count\", \"pl_err\", \"ratio\"] )\n",
    "            df2 = df2.drop(columns=[\"empty\"])\n",
    "            try:\n",
    "                filtered_df2 = df2[df2[\"contigID\"] != \"all.pl_err\"]\n",
    "            except:\n",
    "                filtered_df2=copy.deepcopy(df2)\n",
    "            # Remove the prefix from each entry in 'contigID'\n",
    "            filtered_df2.loc[:, \"contigID\"] = filtered_df2[\"contigID\"].str.replace(\"apples_input_di_mtrx_query_\", \"\", regex=False)\n",
    "            filtered_df2.loc[:, \"contigID\"] = filtered_df2[\"contigID\"].str.replace(\".pl_err\", \"\", regex=False)\n",
    "\n",
    "\n",
    "        combo_df = pd.merge(filtered_df2, df1, how = \"left\", left_on = \"contigID\", right_on = \"genome\")\n",
    "        \n",
    "        # Extract everything before \".part\"\n",
    "        combo_df[\"parent_genome\"] = combo_df[\"contigID\"].str.split(\".part\").str[0]\n",
    "        combo_df_w_trueclass = pd.merge(combo_df, true_clades, how = \"left\", left_on = \"parent_genome\", right_on = \"genome\")\n",
    "        meta_data = d.split (\"_\")\n",
    "        \n",
    "        combo_df_w_trueclass[\"read_err\"] = meta_data[1]\n",
    "        combo_df_w_trueclass[\"model\"] = \"{}_{}\".format(meta_data[2], meta_data[3])\n",
    "\n",
    "\n",
    "        mean_score = combo_df_w_trueclass[\"pl_err\"].mean()\n",
    "        equal_count = (combo_df_w_trueclass[\"top_class\"] == combo_df_w_trueclass[\"clade\"]).sum()/len(combo_df_w_trueclass)\n",
    "\n",
    "        mean_score_rounded = round(mean_score, 3) \n",
    "        equal_count_rounded = round(equal_count, 3) \n",
    "        print (d, mean_score_rounded, equal_count_rounded)\n",
    "\n",
    "        # Append contig length information\n",
    "        col_names = [\"readID\", \"length\", \"A\", \"C\", \"G\", \"T\", \"2\", \"3\", \"4\", \"CpG\", \"tv\", \"ts\", \"CpG-ts\"]\n",
    "        length_df = pd.read_csv(os.path.join(length_folder, \"ONT_reads_best10genomes_TOLqueries_{}_LENGTH.txt\".format(meta_data[1])), sep = \"\\t\", header = None, names = col_names)\n",
    "        length_df[\"length_adj\"] = length_df[\"A\"] + length_df[\"C\"] + length_df[\"G\"] + length_df[\"T\"]\n",
    "        \n",
    "        combo_df_w_trueclass_len = pd.merge(combo_df_w_trueclass, length_df, how=\"left\", left_on = \"contigID\", right_on = \"readID\")\n",
    "\n",
    "        # Concatenate vertically\n",
    "        my_big_df = pd.concat([my_big_df, combo_df_w_trueclass_len], axis=0)\n",
    "        \n",
    "\n",
    "        \"\"\"\n",
    "        # Generate true classes.out output so claded_chunked_TRUE can be tested\n",
    "        if \"Claded\" in d:\n",
    "            # Select only the two columns you want\n",
    "            output_df = combo_df_w_trueclass[[\"genome_x\", \"clade\"]]\n",
    "            output_df = output_df.rename(columns={\"genome_x\": \"genome\"})\n",
    "            output_df = output_df.rename(columns={\"clade\": \"top_class\"})\n",
    "            \n",
    "\n",
    "            # Write to a tab-separated text file without the index\n",
    "            output_df.to_csv(\"true_class_files/true_{}_classes.out\".format(meta_data[1]), sep=\"\\t\", index=False)\n",
    "        \"\"\"\n",
    "\n",
    "# Write to CSV file\n",
    "my_big_df.to_csv(\"ONT_best10TOLquer_varReadErr.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "2f4d4389-b5d5-4513-8b16-22c469185604",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>contigID</th>\n",
       "      <th>count</th>\n",
       "      <th>pl_err</th>\n",
       "      <th>ratio</th>\n",
       "      <th>genome_x</th>\n",
       "      <th>top_class</th>\n",
       "      <th>parent_genome</th>\n",
       "      <th>genome_y</th>\n",
       "      <th>clade</th>\n",
       "      <th>read_err</th>\n",
       "      <th>...</th>\n",
       "      <th>G</th>\n",
       "      <th>T</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>CpG</th>\n",
       "      <th>tv</th>\n",
       "      <th>ts</th>\n",
       "      <th>CpG-ts</th>\n",
       "      <th>length_adj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G000008665.part_1_100</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>G000008665.part_1_100</td>\n",
       "      <td>0</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>13</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>2078</td>\n",
       "      <td>2187</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>996</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G000008665.part_1_102</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>G000008665.part_1_102</td>\n",
       "      <td>0</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>13</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>4401</td>\n",
       "      <td>4727</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1864</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G000008665.part_1_103</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>G000008665.part_1_103</td>\n",
       "      <td>0</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>13</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>2993</td>\n",
       "      <td>3263</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1210</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G000008665.part_1_104</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>G000008665.part_1_104</td>\n",
       "      <td>0</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>13</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>7215</td>\n",
       "      <td>7472</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2628</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G000008665.part_1_106</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>G000008665.part_1_106</td>\n",
       "      <td>0</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>G000008665</td>\n",
       "      <td>13</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>2047</td>\n",
       "      <td>2777</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>792</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1805</th>\n",
       "      <td>G900089075.part_9_94</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>G900089075.part_9_94</td>\n",
       "      <td>0</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>6</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>1682</td>\n",
       "      <td>2606</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>842</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1806</th>\n",
       "      <td>G900089075.part_9_95</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>G900089075.part_9_95</td>\n",
       "      <td>0</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>6</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>2155</td>\n",
       "      <td>4354</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>804</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1807</th>\n",
       "      <td>G900089075.part_9_96</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>G900089075.part_9_96</td>\n",
       "      <td>0</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>6</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>1898</td>\n",
       "      <td>2507</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>674</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1808</th>\n",
       "      <td>G900089075.part_9_99</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>G900089075.part_9_99</td>\n",
       "      <td>0</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>6</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>1342</td>\n",
       "      <td>2035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>658</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1809</th>\n",
       "      <td>G900089075.part_9_9</td>\n",
       "      <td>10069.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>G900089075.part_9_9</td>\n",
       "      <td>0</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>G900089075</td>\n",
       "      <td>6</td>\n",
       "      <td>090</td>\n",
       "      <td>...</td>\n",
       "      <td>1054</td>\n",
       "      <td>1672</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>372</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5274</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1810 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   contigID    count  pl_err   ratio               genome_x  \\\n",
       "0     G000008665.part_1_100  10069.0     3.0  0.0003  G000008665.part_1_100   \n",
       "1     G000008665.part_1_102  10069.0     3.0  0.0003  G000008665.part_1_102   \n",
       "2     G000008665.part_1_103  10069.0     3.0  0.0003  G000008665.part_1_103   \n",
       "3     G000008665.part_1_104  10069.0     2.0  0.0002  G000008665.part_1_104   \n",
       "4     G000008665.part_1_106  10069.0     7.0  0.0007  G000008665.part_1_106   \n",
       "...                     ...      ...     ...     ...                    ...   \n",
       "1805   G900089075.part_9_94  10069.0     2.0  0.0002   G900089075.part_9_94   \n",
       "1806   G900089075.part_9_95  10069.0     5.0  0.0005   G900089075.part_9_95   \n",
       "1807   G900089075.part_9_96  10069.0    13.0  0.0013   G900089075.part_9_96   \n",
       "1808   G900089075.part_9_99  10069.0     6.0  0.0006   G900089075.part_9_99   \n",
       "1809    G900089075.part_9_9  10069.0     7.0  0.0007    G900089075.part_9_9   \n",
       "\n",
       "      top_class parent_genome    genome_y  clade read_err  ...     G     T  2  \\\n",
       "0             0    G000008665  G000008665     13      090  ...  2078  2187  0   \n",
       "1             0    G000008665  G000008665     13      090  ...  4401  4727  0   \n",
       "2             0    G000008665  G000008665     13      090  ...  2993  3263  0   \n",
       "3             0    G000008665  G000008665     13      090  ...  7215  7472  0   \n",
       "4             0    G000008665  G000008665     13      090  ...  2047  2777  0   \n",
       "...         ...           ...         ...    ...      ...  ...   ...   ... ..   \n",
       "1805          0    G900089075  G900089075      6      090  ...  1682  2606  0   \n",
       "1806          0    G900089075  G900089075      6      090  ...  2155  4354  0   \n",
       "1807          0    G900089075  G900089075      6      090  ...  1898  2507  0   \n",
       "1808          0    G900089075  G900089075      6      090  ...  1342  2035  0   \n",
       "1809          0    G900089075  G900089075      6      090  ...  1054  1672  0   \n",
       "\n",
       "      3  4   CpG  tv  ts  CpG-ts  length_adj  \n",
       "0     0  0   996   0   0       0        8504  \n",
       "1     0  0  1864   0   0       0       18275  \n",
       "2     0  0  1210   0   0       0       12923  \n",
       "3     0  0  2628   0   0       0       28614  \n",
       "4     0  0   792   0   0       0        9515  \n",
       "...  .. ..   ...  ..  ..     ...         ...  \n",
       "1805  0  0   842   0   0       0        8549  \n",
       "1806  0  0   804   0   0       0       12826  \n",
       "1807  0  0   674   0   0       0        8978  \n",
       "1808  0  0   658   0   0       0        6707  \n",
       "1809  0  0   372   0   0       0        5274  \n",
       "\n",
       "[1810 rows x 25 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combo_df_w_trueclass_len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "0f730c96-83b1-42b0-b5e7-45e5afafe105",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1860"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(length_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a3a651a-dfba-4af1-bd9f-c01d88c5b339",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55219f36-8f46-45f8-acb6-2b051ab79d10",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate true classes for "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81f5e2bb-e495-4742-a1c2-89ea5feb3e50",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1019a6ae-31f4-4352-98b7-c69acab85f53",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40ef7606-74d1-44de-a2d8-484365edd4a3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec2fdf0d-45b5-4896-8753-f2f8e7cb7b6a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8ea76b7-7906-454c-b294-6233224a872c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae0b3d0b-f232-49fd-a729-2043065393f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "ae4080f7-0c64-4c37-8bfc-3f12f22da32d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge dataframes\n",
    "\n",
    "total_p1 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_Chunked_Model\"\n",
    "total_p2 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_Full_Model\"\n",
    "total_p3 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_Global_Model\"\n",
    "total_p4 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_ChunkedExp_Model\"\n",
    "total_p5 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_ChunkedExp_Global_hybrid_Model\"\n",
    "total_p6 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_ChunkedExp_Global_hybrid_Model\"\n",
    "total_p7 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_Global_Chunk_Model\"\n",
    "total_p8 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v55_8k_s28_clade_All_TOL_CONTIGS_ChunkedExp_Model_v55\"\n",
    "total_p9 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_ChunkedExp_Model_TRUE_CLADE\"\n",
    "total_p10 = \"/Users/nora/Documents/ml_metagenomics/tol_variable_k_resuls/k7_v38_8k_s28_clade_All_TOL_CONTIGS_Full_Model_TRUE_CLADE\"\n",
    "\n",
    "total_f1 = \"summary_FULL_GENOME_CONTIGS.pl_error\"\n",
    "total_f2 = \"summary_FULL_GENOME_CONTIGS_FullModel.pl_error\"\n",
    "total_f3 = \"summary_FULL_GENOME_CONTIGS_GlobalModel.pl_error\"\n",
    "total_f4 = \"summary_FULL_GENOME_CONTIGS_ChunkExpModel.pl_error\"\n",
    "total_f5 = \"summary_FULL_GENOME_CONTIGS_ChunkExp_Global_Model.pl_error\"\n",
    "total_f6 = \"summary_FULL_GENOME_CONTIGS_ChunkExp_Global_Modelp_0.9.pl_error\"\n",
    "total_f7 = \"summary_FULL_GENOME_CONTIGS_GlobalChunkModel.pl_error\"\n",
    "total_f8 = \"summary_FULL_GENOME_CONTIGS_ChunkExpModel_v55.pl_error\" # incomplete classification\n",
    "total_f9 = \"summary_FULL_GENOME_CONTIGS_ChunkExpModel_TRUE_CLADE.pl_error\"\n",
    "total_f10 = \"summary_FULL_GENOME_CONTIGS_FullModel_TRUE_CLADE.pl_error\"\n",
    "\n",
    "\n",
    "\n",
    "t1 = pd.read_csv(os.path.join(total_p1, total_f1), sep=' ', header = 0)\n",
    "t2 = pd.read_csv(os.path.join(total_p2, total_f2), sep=' ', header = 0)\n",
    "t3 = pd.read_csv(os.path.join(total_p3, total_f3), sep=' ', header = 0)\n",
    "t4 = pd.read_csv(os.path.join(total_p4, total_f4), sep=' ', header = 0)\n",
    "t5 = pd.read_csv(os.path.join(total_p5, total_f5), sep=' ', header = 0)\n",
    "t6 = pd.read_csv(os.path.join(total_p6, total_f6), sep=' ', header = 0)\n",
    "t7 = pd.read_csv(os.path.join(total_p7, total_f7), sep=' ', header = 0)\n",
    "t8 = pd.read_csv(os.path.join(total_p8, total_f8), sep=' ', header = 0)\n",
    "t9 = pd.read_csv(os.path.join(total_p9, total_f9), sep=' ', header = 0)\n",
    "t10 = pd.read_csv(os.path.join(total_p10, total_f10), sep=' ', header = 0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f6a70b8c-0c27-461e-9e3d-899a2659cc51",
   "metadata": {},
   "outputs": [],
   "source": [
    "t1[\"model\"] = \"Chunk\"\n",
    "t2[\"model\"] = \"Full\"\n",
    "t3[\"model\"] = \"Global\"\n",
    "t4[\"model\"] = \"ChunkExp\"\n",
    "t7[\"model\"] = \"Global_Chunk\"\n",
    "t8[\"model\"] = \"ChunkExp_v55\"\n",
    "t9[\"model\"] = \"ChunkExp_TrueClade\"\n",
    "t10[\"model\"] = \"Full_TrueClade\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "955ceee0-0dc8-442c-8768-7ed2688f57a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_t = pd.concat([t1, t2, t3, t4, t5, t6, t7, t8, t9, t10], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5b71368a-06c0-453b-b065-d9ef7f69bbd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_t.to_csv(os.path.join(os.getcwd(), \"summary_TOL_full_genome_contigs_six_models.pl_error\"), index=False, sep=' ', header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66b7a0a1-ffd7-44f5-8ce7-465af274d54f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80f50a95-3f00-45c4-9ce0-7e7be960db7f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
